File: server/models/record.js
import { Int32 } from 'mongodb'
import { Schema, model, mongoose } from 'mongoose'

//schema 
const recordSchema = new Schema({
    user_mongo_id: {
        type: Schema.Types.ObjectId,
        required: true,
        trim: true,
        immutable: true
    },

    user_uid: {
        type: String,
        required: true,
        unique: true,
        trim: true,
        immutable: true
    },

    itemName: {
        type: String,
        required: true,
        trim: true,
    },

    date: {
        type: Date,
        required: true,
        default: new Date(Date.now())
    },

    //stored in cents in database
    cost: {
        type: Int32,
        required: true
    },

    //save URL
    photo: {
        type: String,
        required: false,
        default: null
    },

    category: {
        type: String,
        required: false,
        default: null,
        trim: true
    },

    isFavourited: {
        type: Boolean,
        required: true,
        default: false,
        immutable: false
    },

    details: {
        type: String,
        required: false,
        trim: true
    },

    isVisible: {
        type: Boolean,
        required: true,
        default: false,
        immutable: false
    },

    likes: {
        type: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
        required: false,
        default: [],
        immutable: false,
    }
},
    {
        timestamps: true
    }
)

export const Record = model('Record', recordSchema)

File: frontend_munchies/lib/models/record.dart

// ignore_for_file: non_constant_identifier_names

import 'dart:io';

class Record {
  final String? record_id;
  String? user_uid;
  String itemName;
  DateTime date;
  int cost;
  String? photo_URL;
  String? category;
  bool isFavourited;
  String? details;
  bool isVisible;
  File? photo_file;

  Record({
    this.record_id,
    this.user_uid,
    required this.itemName,
    required this.date,
    required this.cost,
    this.photo_URL,
    this.category,
    required this.isFavourited,
    this.details,
    required this.isVisible,
    this.photo_file
  });

  factory Record.fromJson(Map<String, dynamic> json) {
    return Record(
      record_id: json['_id']?.toString() ?? '',
      user_uid: json['user_uid']?.toString() ?? '',
      itemName: json['itemName']?.toString() ?? '',
      date: DateTime.parse(json['date']?.toString() ?? DateTime.now().toIso8601String()).toLocal(),
      cost: int.parse(json['cost']?.toString() ?? "0"),
      isFavourited: bool.parse(json['isFavourited']?.toString() ?? 'false'),
      category: json['category'] ?? 'Other',
      photo_URL: json['photo']?.toString(),
      details: json['details']?.toString(),
      isVisible: bool.parse(json['isVisible']?.toString() ?? 'false')
    );
  }
}